File: Unique_integers_of_sum_zero.py
'''Given an integer n, return any array containing n unique integers such that they add up to 0.

https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/ 

Personal note:
The time complexity of this program is: O(n)
'''

def sumZero(n: int):
    final_list = []
    for i in range(0, n - 1):
        final_list.append(i)
    final_list.append(-sum(final_list))
    print(final_list)


if __name__ == '__main__':
    sumZero(5)

File: Unique_integers_of_sum_zero.java

/*
Given an integer n, return any array containing n unique integers such that they add up to 0.
https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/ 
 */
import java.util.*;
import java.util.stream.*;

class Solution {
	public static int[] sumZero(int n) {
		List<Integer> final_list = new ArrayList<Integer>();
		for (int i = 0; i < n - 1; i++) {
			final_list.add(i);
		}
		int sum = final_list.stream().mapToInt(Integer::intValue).sum();
		final_list.add(-sum);
		int[] array = new int[final_list.size()];
		for (int i = 0; i < final_list.size(); i++) {
			array[i] = final_list.get(i);
		}
		return array;
	}

	public static void main(String[] args) {
		int[] arr = sumZero(7);
		for (int i = 0; i < arr.length; i++) {
			System.out.print(arr[i] + " ");
		}
	}
}
